File: server/src/models/consent_record.model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const ConsentRecord = sequelize.define('ConsentRecord', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  enterprise_id: {
    type: DataTypes.UUID,
    allowNull: false,
  },
  recorded_by: {
    type: DataTypes.UUID,
    allowNull: false, // staff/enumerator who captured it
  },
  consent_version: {
    type: DataTypes.STRING,
    allowNull: false,
    defaultValue: '1.0', // versioned notice
  },
  method: {
    type: DataTypes.ENUM('checkbox', 'verbal', 'signature'),
    allowNull: false,
    defaultValue: 'checkbox',
  },
  is_consented: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  safeguarding_acknowledged: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  signed_at: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  notes: {
    type: DataTypes.TEXT,
    allowNull: true,
  }
}, {
  tableName: 'consent_records',
  timestamps: true,
  underscored: true,
});

module.exports = ConsentRecord;

File: lib/features/workflow/consent/consent_record_model.dart
import 'consent_record_entity.dart';

class ConsentRecordModel extends ConsentRecordEntity {
  const ConsentRecordModel({
    required super.id,
    required super.enterpriseId,
    required super.recordedBy,
    required super.consentVersion,
    required super.method,
    required super.isConsented,
    required super.safeguardingAcknowledged,
    required super.signedAt,
    super.notes,
  });

  factory ConsentRecordModel.fromJson(Map<String, dynamic> json) {
    return ConsentRecordModel(
      id: json['id'] ?? '',
      enterpriseId: json['enterprise_id'] ?? '',
      recordedBy: json['recorded_by'] ?? '',
      consentVersion: json['consent_version'] ?? '1.0',
      method: json['method'] ?? 'checkbox',
      isConsented: json['is_consented'] ?? false,
      safeguardingAcknowledged: json['safeguarding_acknowledged'] ?? false,
      signedAt: DateTime.parse(json['signed_at']),
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'enterprise_id': enterpriseId,
      'recorded_by': recordedBy,
      'consent_version': consentVersion,
      'method': method,
      'is_consented': isConsented,
      'safeguarding_acknowledged': safeguardingAcknowledged,
      'signed_at': signedAt.toIso8601String(),
      'notes': notes,
    };
  }
}
